### PR TITLE
Updated Style Guide with article about formatting code samples

### PR DESCRIPTION
**Pull request by sdwheeler**
This article addresses the use of the PowerShell prompt and the proper use of language labels on code fences.

### CODE REVIEW

File: contributing/FORAMTTING-CODE.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
+# Formatting code blocks
+
+We want to adopt a consistent style for PowerShell code blocks and output in our documentation.
+The existing documentation has uses multiple styles and the formatting rules have changed multiple times.
+The current Open Publishing System (OPS) we use has a syntax highlighting feature that support many languages.
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: "supports"

File: contributing/FORAMTTING-CODE.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+Here is an example of a syntax block:
+
+    ```
+    Get-Command [-Verb <String[]>] [-Noun <String[]>] [-Module <String[]>]
+     [-FullyQualifiedModule <ModuleSpecification[]>] [-TotalCount <Int32>] [-Syntax] [-ShowCommandInfo]
+     [[-ArgumentList] <Object[]>] [-All] [-ListImported] [-ParameterName <String[]>]
+     [-ParameterType <PSTypeName[]>] [<CommonParameters>]
+    ```
+
+This is example is illustrating all of the possible parameters of the `Get-Command` cmdlet.
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: "This example"

File: contributing/FORAMTTING-CODE.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+We want to adopt a consistent style for PowerShell code blocks and output in our documentation.
+The existing documentation has uses multiple styles and the formatting rules have changed multiple times.
+The current Open Publishing System (OPS) we use has a syntax highlighting feature that support many languages.
+OPS also adds a **Copy** button that copies the contents of the code block to the clipboard.
+This allows you to quickly paste the code into a script for testing the code example.
+However, not all examples in our documentation are intended to be run.
+Some code blocks are simple illustrations of a PowerShell concept.
+
+We are defining two scenarios for code blocks:
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: Love the delineation here, hits the nail on the head.

File: contributing/FORAMTTING-CODE.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    PS> "abc", "def" -eq "abc"
+    abc
+    ```
+
+Note that this example has the simplified PowerShell prompt and shows the resulting output.
+In this case, we don't intend the reader to copy this example and try to run the copied code.
+
+## Formatting executable examples
+
+More complex examples or examples that would be useful to copy and execute should use <code>\`\`\`powershell</code> &hellip; <code>\`\`\`</code> code fence markers.
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: Just an FYI, but I believe you can do this like:

```
`` ```powershell `` ... `` ``` ``
```
which will render as: 

`` ```powershell `` ... `` ``` ``

Source: https://gist.github.com/mcandre/8d76e076b2495fd8b36f439ec5033116

File: contributing/FORAMTTING-CODE.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+Note that the regex used only matches simple prompts like "`PS C:\>`".
+Prompts that include subdirectories or paths from other PSDrive providers will not be removed.
+See the general guidelines, in the next section, for the appropriate use of prompts in examples.
+
+## General guidelines
+
+- Code fences
+
+  Markdown allows for indentation to signal code block, however, all code blocks should be contained in a code fence.
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: "to signal" sounds a little weird, maybe "to signify" or "Markdown supports indentation for rendering code blocks"?

File: contributing/FORAMTTING-CODE.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  A code fence is a block of code surrounded by <code>\`\`\`</code> strings.
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: Another opportunity for double backticks if you want 'em

```
`` ``` ``
```

File: contributing/FORAMTTING-CODE.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+  The code fence markers must be on their own line before and after the code sample.
+  The marker at the start of the code block may have an optional language label.
+  See the previous discussion about illustrative and executable examples for the proper use of language labels.
+
+- Line continuation in code samples
+
+  Avoid using line continuation characters (\`) in PowerShell code examples.
+  These are a hard to see and can cause problems if there are extra spaces on the end of the line.
```
Comment (joeyaiello) [inline contributing/FORAMTTING-CODE.md]: "These are hard to see"

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[CONTRIBUTING.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/CONTRIBUTING.md) | :white_check_mark:Succeeded | |
[contributing/FORAMTTING-CODE.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/contributing/FORAMTTING-CODE.md) | :white_check_mark:Succeeded | |
[contributing/STYLE.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/contributing/STYLE.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C14%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201712141759562932-1954%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=pSb3XN03gDWeWDfdvVuJJEDG62AWQM78pi1gxKljSRA%3D&st=2017-12-14T18%3A26%3A43Z&se=2018-01-14T18%3A31%3A43Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: contributing/FORAMTTING-CODE.md (hunk 4 above)
Comment (sdwheeler) [reply contributing/FORAMTTING-CODE.md]: This formatting was submitted by an external contributor and I accepted it then. The use of spaces gets dicey (especially if you get it wrong) and is harder to see the intent of the formatting.

File: contributing/FORAMTTING-CODE.md (hunk 7 above)
Comment (sdwheeler) [reply contributing/FORAMTTING-CODE.md]: Fixed

File: contributing/FORAMTTING-CODE.md (hunk 5 above)
Comment (sdwheeler) [reply contributing/FORAMTTING-CODE.md]: fixed

File: contributing/FORAMTTING-CODE.md (hunk 3 above)
Comment (sdwheeler) [reply contributing/FORAMTTING-CODE.md]: 👍

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[CONTRIBUTING.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/CONTRIBUTING.md) | :white_check_mark:Succeeded | |
[contributing/FORAMTTING-CODE.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/contributing/FORAMTTING-CODE.md) | :white_check_mark:Succeeded | |
[contributing/STYLE.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/contributing/STYLE.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C14%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201712141811079574-1954%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=CmG%2FBNC3n11gWcSKvcAZya%2BFLh2SMhzomtBzpyA4TnI%3D&st=2017-12-14T18%3A56%3A45Z&se=2018-01-14T19%3A01%3A45Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by saldana** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[CONTRIBUTING.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/CONTRIBUTING.md) | :white_check_mark:Succeeded | |
[contributing/FORAMTTING-CODE.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/contributing/FORAMTTING-CODE.md) | :white_check_mark:Succeeded | |
[contributing/STYLE.md](https://github.com/sdwheeler/PowerShell-Docs/blob/sdw-codestyle/contributing/STYLE.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C14%5Cbab27324-4c59-a348-1473-9c6d587e4020%5CPullRequest%5C201712141839454134-1954%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=2s%2FmuGZ72IULMSGM3LQ5n2pIl%2FQptXPbD14YYY554FU%3D&st=2017-12-14T19%3A27%3A28Z&se=2018-01-14T19%3A32%3A28Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**